packages/x-date-pickers/src/locales/arEg.ts

```diff
@@ -0,0 +1,97 @@
+import { PickersLocaleText } from './utils/pickersLocaleTextApi';
+import { getPickersLocalization } from './utils/getPickersLocalization';
+
+// This object is not Partial<PickersLocaleText> because it is the default values
+
+const arEGPickers: PickersLocaleText<any> = {
```

Comment:
We keep every non-English locale `Partial` to avoid having to update it whenever a new entry is added in English locale.
```suggestion
const arEGPickers: Partial<PickersLocaleText<any>> = {
```